### ISSUE TITLE
refresh alert dialog

### ISSUE THREAD

**Issue by 0x00b1** [CellProfiler/CellProfiler#1856]
CellProfiler’s alert dialog is confusing (it has seven buttons and a check box):

<img width="614" alt="pipeline_error" src="https://cloud.githubusercontent.com/assets/315821/13830110/07e067fa-eba1-11e5-9281-dad5bd53c2af.png">

I think it’d be easy to simplify:

<img width="420" alt="blank_skitch_document" src="https://cloud.githubusercontent.com/assets/315821/13830171/5d48e168-eba1-11e5-933c-a41577fd4775.png">

Note:
- We should write informative CellProfiler-specific error messages for common exceptions
- “Stop Processing” is replaced with the canonical “Cancel”
- “Sent Report” is removed entirely. It doesn’t work. Moreover, when we fix this feature, users should globally opt-in or opt-out to reporting (e.g. a check box in preferences)
- While “Skip Image, Continue Pipeline” is useful, it’s better handled from insider the application window
- I don’t know why anyone would want to “Hide details …” 
- Likewise, I don’t know why anyone would want to “Don’t show this error again” if we provide informative error messages
- “Debug in pdb…” is useful to the narrowest group of users: Lee and I. And I rarely use pdb (I prefer IPython and ipdb)
- The dialog should be untitled to match [Apple](https://developer.apple.com/library/mac/documentation/UserExperience/Conceptual/OSXHIGuidelines/) and [Microsoft](https://dev.windows.com/en-us/desktop/design)’s user interface guidelines
- “Copy to clipboard” is easily achieved by selected the text and copying using the appropriate hot key

**Comment by 0x00B1** [comment]
[Guidelines for message dialogs](https://msdn.microsoft.com/en-us/library/windows/apps/hh738363.aspx) (Windows Dev Center)

**Comment by 0x00B1** [comment]
OS X

```Swift
import Cocoa

@NSApplicationMain
class AppDelegate: NSObject, NSApplicationDelegate {
  @IBOutlet weak var window: NSWindow!

  func applicationDidFinishLaunching(aNotification: NSNotification) {
    let alert = NSAlert()
    
    alert.messageText = "Morbi vitae viverra ex"
    
    alert.informativeText = "Aliquam fringilla interdum nunc, vitae efficitur arcu bibendum ac. Vestibulum non risus."
    
    alert.icon = NSImage(named: NSImageNameCaution)
    
    alert.addButtonWithTitle("Continue Processing")
    alert.addButtonWithTitle("Cancel")
    
    let result = alert.runModal()
    
    switch(result) {
    case NSAlertFirstButtonReturn:
      print("continued")
    case NSAlertSecondButtonReturn:
      print("canceled")
    default:
      break
    }
  }

  func applicationWillTerminate(aNotification: NSNotification) {

  }
}
```

**Comment by 0x00B1** [comment]
wxPython prototype

```Python
import sys
import wx


class Frame(wx.Frame):
    def __init__(self):
        super(Frame, self).__init__(parent=None, title="Exceptions")

        sys.excepthook = excepthook

        panel = Panel(self)

        self.Show()


class MessageDialog(wx.MessageDialog):
    def __init__(self, message, extended_message):
        super(MessageDialog, self).__init__(parent=None, message=message, style=wx.CANCEL | wx.ICON_EXCLAMATION)

        self.SetExtendedMessage(extended_message)

        self.SetOKLabel("Continue Processing")


class Panel(wx.Panel):
    def __init__(self, parent):
        super(Panel, self).__init__(parent)

        button = wx.Button(self, label="Raise Exception")

        button.Bind(wx.EVT_BUTTON, self.throw)

    @staticmethod
    def throw(event):
        return 1 / 0


def excepthook(exception, message, backtrace):
    message = "Morbi vitae viverra ex"

    extended_message = "Aliquam fringilla interdum nunc, vitae efficitur arcu bibendum ac. Vestibulum non risus."

    dialog = MessageDialog(message, extended_message)

    dialog.ShowModal()

    dialog.Destroy()


if __name__ == "__main__":
    application = wx.App(False)

    frame = Frame()

    application.MainLoop()
```

**Comment by 0x00B1** [comment]
wxPython prototype with Sentry integration for logging exceptions

```Python
import raven
import sys
import wx


class Client(raven.Client):
    def __init__(self):
        dsn = "https://3d53494dbaaf4e858afd79f56506a749:8a7a767a1924423f89c1fdfd69717fd5@app.getsentry.com/70887"

        super(Client, self).__init__(dsn)


class Frame(wx.Frame):
    def __init__(self):
        super(Frame, self).__init__(parent=None, title="Exceptions")

        self.sentry = Client()

        sys.excepthook = self.excepthook

        Panel(self)

        self.Show()

    def excepthook(self, *exc_info):
        self.sentry.captureException(exc_info)

        message = "Morbi vitae viverra ex"

        extended_message = "Aliquam fringilla interdum nunc, vitae efficitur arcu bibendum ac. Vestibulum non risus."

        dialog = MessageDialog(message, extended_message)

        dialog.ShowModal()

        dialog.Destroy()


class MessageDialog(wx.MessageDialog):
    def __init__(self, message, extended_message):
        super(MessageDialog, self).__init__(parent=None, message=message, style=wx.CANCEL | wx.ICON_EXCLAMATION)

        self.SetExtendedMessage(extended_message)

        self.SetOKLabel("Continue Processing")


class Panel(wx.Panel):
    def __init__(self, parent):
        super(Panel, self).__init__(parent)

        button = wx.Button(self, label="Raise Exception")

        button.Bind(wx.EVT_BUTTON, self.throw)

    @staticmethod
    def throw(event):
        return 1 / 0


if __name__ == "__main__":
    application = wx.App(False)

    frame = Frame()

    application.MainLoop()
```

**Comment by 0x00B1** [comment]
I think this issue is ready for consideration. In my opinion:

* It simplifies the current implementation. errordialog.py is 500 SLOC, my implementation is 22 SLOC.

* It conforms to platform usability guidelines. This is important from both an accessibility and user experience perspective.

* It includes a robust mechanism for real-time crash reporting.

* It’s prettier. :tulip: 

What does everyone think?

**Comment by jhung0** [comment]
Nice! It's a really good idea to simplify the error messages in the way Allen showed (and fix reporting of them). The important things to me are that common error messages come with suggestions for the user, make reporting errors become easier, and make the error window look less scary.

**Comment by daviddao** [comment]
+1 on my side.

**Comment by dlogan** [comment]
I like it!  

Agreed with Jane on all points
- _Simplified and less scary dialogs_.  This seems to be a vast improvement.
- _Make reporting easier_.  Sentry seems to do that nice and unobtrusively.  The only thing I'd add here (or maybe it's already coded?) is that it might be useful to add an opt-in on the first error for the user, which sets the pref fro then on
- _Suggestions for the user_.  Yes!  Just my gut feelings, and perhaps difficult to do for all errors, but we should attempt to suggest perhaps one of the following: (a) Don't continue, (b) Continue is likely fine, or (c) Safe to continue in Test Mode for display, but don't trust measurements.  
At least, that's my thought process when encountering an exception
- "Cancel" is not clear on this dialog.  Cancel the dialog only? Cancel the processing?  I like the "Stop processing" nomenclature.

**Comment by AnneCarpenter** [comment]
Wow, this issue is a work of art. 

I concur with the enthusiastic responses and the details. 

I also think "Stop processing" is more clear to the user but wouldn't cry if you refuse. 

I don't understand what you meant by "it’s better handled from insider the application window" for   “Skip Image, Continue Pipeline” - can you elaborate? I don't see how you can have that functionality anywhere but within the dialog.

I agree “Copy to clipboard” SHOULD be easily achieved by selecting the text and copying using the appropriate hot key but it doesn't work in the current form, hence the button. If you can make it work then YES jettison the button!

**Comment by mcquin** [comment]
Resolved #2663
